src/gui/processing/qgsprocessingmeshdatasetwidget.cpp

```diff
@@ -0,0 +1,568 @@
+/***************************************************************************
+  qgsprocessingmeshdatasetgroupswidget.h
+  ---------------------
+  Date                 : October 2020
+  Copyright            : (C) 2020 by Vincent Cloarec
+  Email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsprocessingmeshdatasetwidget.h"
+#include "qgsdatetimeedit.h"
+#include "qgsprocessingmultipleselectiondialog.h"
+#include "qgsmeshlayer.h"
+#include "qgsmeshlayertemporalproperties.h"
+#include "qgspanelwidget.h"
+#include "qgsmapcanvas.h"
+
+#include <QLineEdit>
+#include <QLabel>
+#include <QMenu>
+#include <QToolButton>
+#include <QVBoxLayout>
+
+/// @cond PRIVATE
+
+QgsProcessingMeshDatasetGroupsWidget::QgsProcessingMeshDatasetGroupsWidget( QWidget *parent, const QgsProcessingParameterMeshDatasetGroups *param )
+  : QWidget( parent ),
+    mParam( param )
+{
+  QHBoxLayout *hl = new QHBoxLayout();
+  hl->setContentsMargins( 0, 0, 0, 0 );
+
+  mLineEdit = new QLineEdit();
+  mLineEdit->setEnabled( false );
+  hl->addWidget( mLineEdit, 1 );
+
+  mToolButton = new QToolButton();
+  mToolButton->setText( QString( QChar( 0x2026 ) ) );
+  hl->addWidget( mToolButton );
+
+  setLayout( hl );
+
+  mLineEdit->setText( tr( "%1 dataset groups selected" ).arg( 0 ) );
+
+  mToolButton->setPopupMode( QToolButton::InstantPopup );
+  QMenu *toolButtonMenu = new QMenu( this );
+  connect( toolButtonMenu->addAction( tr( "Current active dataset group" ) ),
+           &QAction::triggered, this, &QgsProcessingMeshDatasetGroupsWidget::selectCurrentActiveDatasetGroup );
+  connect( toolButtonMenu->addAction( tr( "Select in available dataset groups" ) ),
```

Comment:
```suggestion
  connect( toolButtonMenu->addAction( tr( "Select in Available Dataset Groups" ) ),
```

although this wording sounds strange to me -- should it be "select FROM available data groups"?